modules/oxide-components/src/main/ts/keynav/keyboard/FlowType.stories.tsx

```diff
@@ -0,0 +1,289 @@
+    background-color: #cadbee;
+  }
+  .skip:focus {
+    background-color: red;
+  }
+`;
+
+const store = TestStore();
+
+// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
+const meta = {
+  title: 'KeyboardNavigationHooks/FlowKey',
+  component: FlowTypeDemo,
+  parameters: {
+    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
+    layout: 'centered',
+    docs: {
+      story: {
+        autoplay: true
+      }
+    }
+  },
+  tags: [ 'autodocs', 'skip-visual-testing' ],
+  decorators: [
+    (Story: PartialStoryFn<ReactRenderer>): ReactElement => (
+      <>
+        <style>
+          {styles}
+        </style>
+        <Story />
+      </>
+    )
+  ],
+  beforeEach: async () => {
+    store.clear();
+  },
+  args: {
+    selector: '.stay',
+  },
+  argTypes: {
+    containerRef: {
+      control: false,
+      type: {
+        required: true,
+        name: 'other',
+        value: 'RefObject<HTMLElement>'
+      },
+      description: 'RefObject<HTMLElement> - Reference to the container element that will handle keyboard navigation',
+      table: {
+        type: { summary: 'RefObject<HTMLElement>' },
+        defaultValue: { summary: 'useRef<HTMLDivElement>(null)' },
+      },
+    }
+  },
+  play: ({ canvasElement, context }) => {
+    const container = canvasElement.ownerDocument.querySelector('.container');
+    if (container) {
+      FocusTools.setFocus(SugarElement.fromDom(container), context.args.selector);
+    }
+  },
+} satisfies Meta;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const Basic: Story = {};
+
+interface ItemProps {
+  classes: string[];
+  name: string;
+  store: TestStore;
+}
+
+const Item = ({ classes, name, store }: ItemProps) =>
+  (
+    <span
+      className={classes.join(' ')}
+      style={{
+        display: 'inline-block',
+        width: '20px',
+        height: '20px',
+        margin: '2px',
+        border: '1px solid ' + (classes.includes('stay') ? 'blue' : 'yellow')
+      }}
+      tabIndex={-1}
+      onClick={store.adder('item.execute: ' + name)}
+    />
+  );
+
+const Container = forwardRef<HTMLDivElement, { store: TestStore; testId: string }>(({ store, testId }, ref) => (
+  <div
+    data-testid={testId}
+    ref={ref}
+    className="flow-keying-test"
+    style={{
+      background: 'white',
+      width: '200px',
+      height: '200px'
+    }}
+  >
+    <Item classes={[ 'stay', 'one' ]} name="one" store={store} />
+    <Item classes={[ 'stay', 'two' ]} name="two" store={store} />
+    <Item classes={[ 'skip', 'three' ]} name="three" store={store} />
+    <Item classes={[ 'skip', 'four' ]} name="four" store={store} />
+    <Item classes={[ 'stay', 'five' ]} name="five" store={store} />
+  </div>
+));
+
+const sequence = async (doc: Document, key: string, identifiers: Array<{ label: string; selector: string }>) => {
```

Comment:
I wonder if we should start having utility modules.. This seems like a generic function that we might want to reuse for other tests.